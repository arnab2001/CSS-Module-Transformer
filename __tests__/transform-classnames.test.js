const babel = require('@babel/core');
const plugin = require('./../src/transform-classname');

const transformCode = (code) => {
  return babel.transform(code, {
    plugins: [plugin],
  }).code;
};

describe('Babel Plugin Transform Classnames', () => {
  it('transforms simple static class name', () => {
    const input = '<div className="example" />';
    const output = transformCode(input);
    expect(output).toMatchSnapshot();
  });

  it('transforms dynamic class name with hyphen', () => {
    const input = '<div className={dynamicClassName} />';
    const output = transformCode(input);
    expect(output).toMatchSnapshot();
  });

  it('does not transform dynamic class name without hyphen', () => {
    const input = '<div className={dynamicClassName} />';
    const output = transformCode(input);
    expect(output).toMatchSnapshot();
  });

  it('leaves other JSX attributes unchanged', () => {
    const input = '<div id="myDiv" data-custom="value" />';
    const output = transformCode(input);
    expect(output).toMatchSnapshot();
  });

  it('transforms CSS file import', () => {
    const input = 'import "./styles.css";';
    const output = transformCode(input);
    expect(output).toMatchSnapshot();
  });
});
