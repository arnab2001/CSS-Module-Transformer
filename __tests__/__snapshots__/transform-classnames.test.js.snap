// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Babel Plugin Transform Classnames does not transform dynamic class name without hyphen 1`] = `
"\\"use strict\\";

/*#__PURE__*/React.createElement(\\"div\\", {
  className: dynamicClassName
});"
`;

exports[`Babel Plugin Transform Classnames leaves other JSX attributes unchanged 1`] = `
"\\"use strict\\";

/*#__PURE__*/React.createElement(\\"div\\", {
  id: \\"myDiv\\",
  \\"data-custom\\": \\"value\\"
});"
`;

exports[`Babel Plugin Transform Classnames transforms CSS file import 1`] = `
"\\"use strict\\";

require(\\"./styles.module.css\\");"
`;

exports[`Babel Plugin Transform Classnames transforms dynamic class name with hyphen 1`] = `
"\\"use strict\\";

/*#__PURE__*/React.createElement(\\"div\\", {
  className: dynamicClassName
});"
`;

exports[`Babel Plugin Transform Classnames transforms simple static class name 1`] = `
"\\"use strict\\";

/*#__PURE__*/React.createElement(\\"div\\", {
  className: \\"example\\"
});"
`;
